homeassistant/components/homekit/util.py

```diff
@@ -0,0 +1,66 @@
+"""Collection of useful functions for the HomeKit component."""
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.core import split_entity_id
+from homeassistant.const import (
+    ATTR_CODE)
+import homeassistant.helpers.config_validation as cv
+from .const import (
+    HOMEKIT_NOTIFY_ID, QR_CODE_NAME)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_entity_config(values):
+    """Validate config entry for CONF_ENTITY."""
+    entities = {}
+    for key, config in values.items():
+        entity = cv.entity_id(key)
+        params = {}
+        if not isinstance(config, dict):
+            raise vol.Invalid('The configuration for "{}" must be '
+                              ' an dictionary.'.format(entity))
+
+        domain, _ = split_entity_id(entity)
+
+        if domain == 'alarm_control_panel':
+            code = config.get(ATTR_CODE)
+            params[ATTR_CODE] = cv.string(code) if code else None
+
+        entities[entity] = params
+    return entities
+
+
+def show_setup_message(bridge, hass):
+    """Display persistent notification with setup information."""
+    pin = bridge.pincode.decode()
+    path = hass.config.path('www/' + QR_CODE_NAME)
+    try:
+        bridge.qr_code.png(path, scale=12, quiet_zone=2,
```

Comment:
Don't write it to disk. Register an HTTP view to serve the QR code and generate it on the fly.

That way you don't have to worry about cleaning up the file (as in its current state it can be easily left undeleted).